### PR TITLE
minimal mpc in the loop

### PR DESCRIPTION
**Pull request by yangwill**
Large refactor (moving files around/moving gains to different places)

Main new features
- new diagram: `centroidal_mpc_reference.cc`: takes in robot state and joystick commands and outputs centroidal mpc solution
- new diagram: `run_mpc_visualization.cc`: takes in centroidal mpc solution and draws traces of trajectory in drake visualizer and meshcat
- `cassie_kcmpc_trajopt`: now all the settings are contained in yamls: motion specifications are contained in `examples/Cassie/kinematic_centroidal_planner/gaits` and `examples/Cassie/kinematic_centroidal_planner/motions`. Where motions contains whole trajectory parameters including the gait sequence.
- small bazel things like combining data files into `filegroup` rule

- - -
This change is [Reviewable](https://reviewable.io/reviews/dairlab/dairlib/329)

### CODE REVIEW

File: systems/controllers/kinematic_centroidal_mpc/kinematic_centroidal_mpc.cc [hunk 1]
```diff
@@ -1,656 +1,217 @@
 #include "kinematic_centroidal_mpc.h"
 
-#include <iostream>
-#include <utility>
+#include <drake/common/yaml/yaml_io.h>
```
[the remark this answers is not in the document]
Comment (ShaneRozenLevy) [reply systems/controllers/kinematic_centroidal_mpc/kinematic_centroidal_mpc.cc:3]: Looking more closely, my guess is this class will change significantly when we start doing actual mpc, so I'm fine to save leave it as is for now.

**Review by ShaneRozenLevy — COMMENTED** [review]
Looks close. I have some questions about the flag `KCMPC_REF` name and the code duplication between `cassie_kcmpc_trajopt` and `kinematic_centroidal_mpc.*`, but otherwise seems good.

File: systems/controllers/kinematic_centroidal_mpc/kinematic_centroidal_mpc.cc [hunk 2]
```diff
@@ -215,7 +215,7 @@ void KinematicCentroidalMPC::CalcTraj(
 
   std::cout << "Solving optimization\n\n";
   solver_->Solve();
-  traj->saved_traj = solver_->GenerateLcmTraj(n_knot_points_);
+  traj->saved_traj = solver_->GenerateLcmTraj(n_knot_points_, robot_output->get_timestamp() + 1.0);
```
Comment (ShaneRozenLevy) [inline systems/controllers/kinematic_centroidal_mpc/kinematic_centroidal_mpc.cc:218]: whats the `+1` here for?

**Review by ShaneRozenLevy — APPROVED** [review]
Looks good. Minor question about a magic number.

**Review by yangwill — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/dairlab/dairlib/329)* status: 0 of 116 files reviewed, 10 unresolved discussions (waiting on @ShaneRozenLevy)

___
*[`systems/controllers/kinematic_centroidal_mpc/kinematic_centroidal_mpc.cc` line 218 at r4](https://reviewable.io/reviews/dairlab/dairlib/329#-NH68JPEQchv1jj_INp0-r4-218:-NIoge8cEDwnQWAha5-D:b-gdou45) ([raw file](https://github.com/dairlab/dairlib/blob/b231f1161b45e86ecb2e62fe11c86c0e57297666/systems/controllers/kinematic_centroidal_mpc/kinematic_centroidal_mpc.cc#L218)):*
<details><summary><i>Previously, ShaneRozenLevy wrote…</i></summary><blockquote>

whats the `+1` here for?
</blockquote></details>

removed the magic number for now, added TODO to remind us

**Pull request merged**